### PR TITLE
#165998462 Add dropdown for statistics

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
#### What does this PR do?
- This PR adds a dropdown in the statistics section to enable different time periods to be selected for statistics to be carried out.

#### Description of Task to be completed?
- Add a dropdown with different time periods
- Update the statistics route to include a variable in which a time period is passed to the backend.

#### How should this be manually tested?
1. Git clone the repository.
2. Checkout to the `ft-statistics-dropdown-165998462` branch.
5. Start the server `npm run start` _(or the `yarn` equivalent)_.
6. Clone the backend repository, `https://github.com/andela/bp-esa-backend`.
7. Checkout to the backend `ft-add-statistics-period-165998462` branch.
8. Start the backend server `npm run start:dev` _(or the `yarn` equivalent)_.
9. Login to the ESA application
10. Click the caret in the statistics section.
11. Click a time period.
The statistics for the selected time period should be displayed.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[165998462](https://www.pivotaltracker.com/story/show/165998462)

#### Screenshots (If applicable)
Screen Shot 2019-05-16 at 15 44 56

### CODE REVIEW

**Review by mikenthiwa — APPROVED** [review]
Works as expected. LGTM

**Comment by mikenthiwa** [comment]
@MuhanguziDavid Fix merge conflicts.

**Comment by MuhanguziDavid** [comment]
Merge conflicts fixed. Thanks

File: src/components/StatsCard/styles.scss [hunk 1]
```diff
@@ -63,3 +63,41 @@
...
+    min-width: 100px;
+    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
+    border-radius: 5px;
+    z-index: 1;
+    margin: 1% 0 0 73%;
+    text-align: left;
+    font-size: medium;
+
+    .stat-dropdown-item {
+      padding-left: 7px;
```
Comment (09arnold) [inline src/components/StatsCard/styles.scss]: change this to `padding: 5px 10px;`
that'll look better

**Comment by 09arnold** [comment]
The dropdown does not close when it loses focus, ie, when you click outside of it. Please fix that

File: src/components/StatsCard/styles.scss (hunk 1 above)
Comment (MuhanguziDavid) [reply src/components/StatsCard/styles.scss]: Changes have been made. Thanks

**Comment by MuhanguziDavid** [comment]
Thanks @09arnold , I have added functionality to close the dropdown when one clicks outside of it.

File: src/components/StatsCard/styles.scss [hunk 2]
```diff
@@ -63,3 +63,41 @@
...
+  .stat-dropdown-caret {
+    border: 1px solid #CCD;
+    border-radius: 5px;
+    margin-left: 4px;
+    padding-right: 4px;
+  }
+
+  .stat-dropdown-caret:hover {
+    cursor: pointer;
+    background-color: #FCFCFD;
```
Comment (09arnold) [inline src/components/StatsCard/styles.scss]:
```suggestion
```

File: src/components/StatsCard/styles.scss [hunk 3]
```diff
@@ -63,3 +63,41 @@
     margin: auto;
   }
 }
+
+.stat-dropdown-container {
+  position: relative;
+  display: inline-block;
+
+  .stat-dropdown-caret {
+    border: 1px solid #CCD;
```
Comment (09arnold) [inline src/components/StatsCard/styles.scss]:
```suggestion
```

**Review by 09arnold — APPROVED** [review]
Please remove the items in the styles as suggested

File: src/components/StatsCard/styles.scss (hunk 3 above)
Comment (MuhanguziDavid) [reply src/components/StatsCard/styles.scss]: @09arnold , this border was requested by @matthewacha . I have removed it tho.

**Comment by MuhanguziDavid** [comment]
@09arnold , I have removed the items in the styles as suggested.

**Pull request merged**